From f548589f79418dd98c0b95b0d1375484a0070c8a Mon Sep 17 00:00:00 2001
From: Jingnan Zhu <e3334c@motorola.com>
Date: Sat, 26 Mar 2016 14:11:52 -0500
Subject: [PATCH 828/959] IKSWM-27957:input: misc: vl53l0: limit reporting freq
 in super mode

there has too many update of the status when in super mode to sensor

Change-Id: Ic5390c04d7dc73a833ae28ca7fa3424286cc9456
Signed-off-by: Jingnan Zhu <e3334c@motorola.com>
Reviewed-on: http://gerrit.mot.com/837785
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/input/misc/vl53L0/stmvl53l0_module.c | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/drivers/input/misc/vl53L0/stmvl53l0_module.c b/drivers/input/misc/vl53L0/stmvl53l0_module.c
index 01321a7436a..67c3861181d 100644
--- a/drivers/input/misc/vl53L0/stmvl53l0_module.c
+++ b/drivers/input/misc/vl53L0/stmvl53l0_module.c
@@ -60,7 +60,8 @@ struct timeval start_tv, stop_tv;
 static char *devname = "laser";
 
 static struct stmvl53l0_data *gp_vl53l0_data;
-
+static int s_count;
+static int s_prevalue;
 
 static struct stmvl53l0_module_fn_t stmvl53l0_module_func_tbl_cci = {
 	.init = stmvl53l0_init_cci,
@@ -988,8 +989,8 @@ static void stmvl53l0_enter_sar(struct stmvl53l0_data *data, uint8_t from)
 }
 static void stmvl53l0_enter_super(struct stmvl53l0_data *data, uint8_t from)
 {
-	vl53l0_dbgmsg_en("Enter, stmvl53l0_enter_super flag:%d\n", from);
-	if (from == SAR_MODE) {
+	vl53l0_errmsg("Enter, stmvl53l0_enter_super flag:%d\n", from);
+	if (from == SAR_MODE || from == SUPER_MODE) {
 		vl53l0_dbgmsg_en("Call of VL53L0_DEVICEMODE_CONTINUOUS_RANGING\n");
 		papi_func_tbl->ClearInterruptMask(data, 0);
 
@@ -1209,6 +1210,7 @@ VL53L0_RangingMeasurementData_t	   RMData;
 struct stmvl53l0_data *data = gp_vl53l0_data;
 VL53L0_DEV vl53l0_dev = data;
 struct timeval tv;
+int newv;
 
 do_gettimeofday(&tv);
 papi_func_tbl->GetRangingMeasurementData(vl53l0_dev, &RMData);
@@ -1293,8 +1295,14 @@ wake_up(&data->range_data_wait);
 		vl53l0_dbgmsg_en("SAR measurement ready\n");
 	} else if (SUPER_MODE == data->w_mode) {
 		vl53l0_dbgmsg_en("SUPER_MODE\n");
-		stmvl53l0_ps_read_measurement(vl53l0_dev);
 		stmvl53l0_apply_parameters(data);
+		newv = data->rangeData.RangeMilliMeter > data->highv ? 100:10;
+		if (s_prevalue != newv || s_count > 100)
+			stmvl53l0_ps_read_measurement(vl53l0_dev);
+		if (s_count > 100)
+			s_count = 0;
+		s_count++;
+		s_prevalue = newv;
 		papi_func_tbl->StartMeasurement(data);
 	}
 
-- 
2.11.0

